ui/src/message_popup/mobile_prototype/text_footer_component.jsx

```diff
@@ -0,0 +1,151 @@
+      initialMessageSent: false,
+    });
+  },
+  
+  onTextChange(event, value){
+    this.setState({text: value});
+  },
+  
+  onSendMessage(){
+    this.props.setInitialResponse(this.state.text);
+    this.setState({initialMessageSent: true});
+    this.props.log.logResponse();
+  },
+  
+  onRevisionSendMessage(){
+    this.props.setRevisedResponse(this.state.text);
+    this.props.log.logRevision();
+  },
+  
+  onPassSendMessage(){
+    this.props.setPassedResponse('');
+    this.props.log.logRevisionDeclined();
+  },
+  
+  onNextQuestion(){
+    this.props.onQuestionDone();
+  },
+  
+  render(){
+    const mainButtonLabel = this.renderButtonText();
+    return (
+      <div>
+        <Divider />
+        <div style={styles.responseTextBox}>
+          <TextField 
+            id="Response"
+            multiLine={true}
+            underlineShow={false}
+            rows={2}
+            rowsMax={2}
+            onChange={this.onTextChange}
+            disabled={this.props.isReadyToMoveOn}
+            value={this.state.text === undefined ? '' : this.state.text}
+            hintText="Type something you would say"
+            style={styles.responseTextField}
+            textareaStyle={styles.responseTextAreaInner}
+            />
+        </div>
+        <div style={styles.responseButtonRow}>  
+          <div>
+            <RaisedButton
+              style={styles.responseButton}
+              label={mainButtonLabel}
+              primary={this.props.isReadyToMoveOn}
+              secondary={!this.props.isReadyToMoveOn}
+              onTouchTap={this.props.isReadyToMoveOn ? this.onNextQuestion : this.state.initialMessageSent ?  this.onRevisionSendMessage : this.onSendMessage}
+              disabled={(this.state.text === undefined || this.state.text === '') && !this.props.isReadyToMoveOn}
+              />
+            {this.props.helpType === 'feedback' && this.state.initialMessageSent && !this.props.isReadyToMoveOn &&
+            <RaisedButton
+              style={styles.responseButton}
+              label="Pass"
+              onTouchTap={this.onPassSendMessage}
+              />
+            }
+            {this.props.helpType === 'hints' && !this.props.isReadyToMoveOn &&
+            <RaisedButton
+              style={styles.responseButton}
+              label="Show Example"
+              disabled={this.props.nextExample(false) === null}
+              onTouchTap={this.props.onShowExampleClicked}
+              />
+            }
+          </div>
+          <div style={styles.responseTimer}>
+            {this.props.elapsedMs/1000 + 's'}
+          </div>
+        </div>
+      </div>
+    );
+  },
+
+  renderButtonText(){
+    if(this.props.isReadyToMoveOn) return this.props.nextButtonLabel;
+    if(this.props.helpType === 'feedback' && this.state.initialMessageSent) return 'Revise response';
+    return `Respond to ${this.props.mainStudent !== undefined ? this.props.mainStudent.name : 'Class'}`;
+  }
```

Comment:
@david-rosales This looks good to me!  And a good place where setting up tests will be helpful :)

Formatting the code from https://github.com/mit-teaching-systems-lab/threeflows/pull/83/commits/b019b831572b99e47bdc5ec447dd47f7e88a1a9a, the bug came up when the user is first shown the question in solution mode, before they've typed anything:

```
const mainButtonLabel = this.props.isReadyToMoveOn
  ? this.props.nextButtonLabel
  : this.props.helpType !== 'feedback'
    ? 'Respond to ' + (this.props.mainStudent !== undefined ? this.props.mainStudent.name : 'Class')
    : this.state.initialMessageSent
      ? 'Revise Response'
      : 'Save Response';
```

looks good now!